src/components/BannerCloseButton.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useEffect } from 'react';
+import { css } from '@emotion/react';
+import { space, from } from '@guardian/source-foundations';
+import { Button, SvgCross } from '@guardian/source-react-components';
+import type { TrackClick } from '../utils/tracking';
+import type { CloseButtonColorStyles } from '../styles/colorData';
+
+// Pass a function to run when the user hits the escape key
+// Useful for enabling a keyboard shortcut for dismissing banners
+export function useEscapeShortcut(
+    eventHandler: (event: KeyboardEvent) => void,
+    deps: React.DependencyList = [],
+): void {
+    function handleEscapeKeydown(event: KeyboardEvent) {
+        // IE key name is 'Esc', because IE
+        const isEscapeKey = event.key === 'Escape' || event.key === 'Esc';
+        if (isEscapeKey) {
+            eventHandler(event);
+        }
+    }
+
+    useEffect(() => {
+        window.addEventListener('keydown', handleEscapeKeydown);
+
+        return () => window.removeEventListener('keydown', handleEscapeKeydown);
+    }, deps);
+}
+
+export type OnCloseClick = (
+    evt: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    internalButtonId: number,
+) => void;
+
+export const CLOSE_BUTTON_ID = 1;
+
+type BannerCloseButtonProps = {
+    trackClick: TrackClick;
+    setShowBanner: (val: boolean) => void;
+    ophanComponentId?: string;
+    colors?: CloseButtonColorStyles;
+};
+
+export const BannerCloseButton: React.FC<BannerCloseButtonProps> = (
+    props: BannerCloseButtonProps,
+) => {
+    const {
+        trackClick,
+        setShowBanner,
+        ophanComponentId,
+        colors = defaultBannerCloseButtonColors,
```

Comment:
AFAICT the defaults are never used since the calling components always pass these (which I think makes sense - I wonder if it's possible to have default colours since this is used in different contexts). Perhaps the default should be removed and the prop becomes required?

Reply:
Good spot. All the static banners should be using the default colors and thus don't need to pass them back into the component. This has been fixed.